bin/scaffold.js

```diff
@@ -0,0 +1,223 @@
+        })
+        .then(function (entry) {
+            return entry.getTree();
+        })
+        .then(function (tree) {
+            var entries = tree.entries()
+                .filter(function (entry) {
+                    return Path.extname(entry.toString()) === '.js';
+                });
+
+            var blobs = entries
+                .map(function (entry) {
+                    return entry.getBlob();
+                });
+
+            var names = entries
+                .map(function (entry) {
+                    return Path.basename(entry.toString(), '.js');
+                });
+
+            return Bluebird.all(blobs)
+                .map(function (blob, index) {
+                    var name = names[index];
+                    var jsdoc = JsdParse(blob.toString());
+
+                    return {
+                        name: name,
+                        description: jsdoc[0] ? jsdoc[0].description : 'no description.'
+                    };
+                });
+        })
+        .then(function (scaffolds) {
+            console.log('Available webtasks:'.green.bold);
+
+            scaffolds
+                .forEach(function (scaffold) {
+                    console.log(scaffold.name.bold.white + ':', scaffold.description.grey);
+                });
+        });
+}
+
+function scaffoldFrom(argv, commit, path) {
+    var filename = argv.params.name ? (argv.params.name + '.js') : 'webtask.js';
+
+    return commit
+        .getEntry(path)
+        .then(function (entry) {
+            return entry.getBlob();
+        })
+        .then(function (blob) {
+            return WriteFile(filename, blob.toString())
+                .then(function () {
+                    return blob;
+                });
+        })
+        .then(function (blob) {
+            return GetTaskCfg(argv, blob.toString());
+        })
+        .then(function (cfg) {
+            var create_args = [
+                'create',
+                filename,
+                '-n',
+                argv.params.name || Path.basename(path, '.js'),
+                '-p',
+                argv.profile
+            ];
+
+            if(cfg.param)
+                addArgumentsTo(create_args, cfg.param, '--param');
+
+            if(cfg.secret)
+                addArgumentsTo(create_args, cfg.secret, '--secret');
+
+            return ExecFile(__dirname + '/wt', create_args);
+        })
+        .then(function (output) {
+            var url = output[0].slice(0, -1);
+
+            console.log('Scaffold written to'.blue, filename.bold.green);
+            console.log('Scaffold deployed to'.blue, url.bold.green)
+        });
+}
+
+function cleanup() {
+    Rimraf.sync('.tmp');
+}
+
+function addArgumentsTo(args, obj, type) {
+    Object.keys(obj)
+        .forEach(function (key) {
+            var hash = [key, obj[key]].join('=');
+
+            args.push(type, hash);
+        });
+}
+
+function getCommit(repo, branch) {
```

Comment:
I think we should be able to improve on this in the future to avoid the need to actually pull down the repo. (For a future PR).